app/bundles/CoreBundle/Queue/QueueService.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2016 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\CoreBundle\Queue;
+
+use Mautic\CoreBundle\Helper\CoreParametersHelper;
+use Mautic\CoreBundle\Queue\RabbitMq;
+use Mautic\CoreBundle\Queue\QueueProtocol;
+
+
+// register here the different types of queue protocols
+abstract class QueueProtocol
```

Comment:
Per Symfony coding standards, each of these abstract classes would need to be in their own files. However, I'd rather see these converted to events to support more of "drop in" solutions via subscribers rather than hard coding the support. This would also pave the way to support other requests beside email and page hits. 

Ideally, I'd like to see this converted to a QueueBundle just to organize the feature. 

Looking at the code, I see three new events:

1. Event dispatched in the queue bundle's config subscriber to collect supported protocols and their config settings that can be injected into the Configuration's Queue section. I'd like to see a select with the list of available protocols (Disabled, RabitMq, Beanstalkd, etc). When one is selected, it's applicable configuration settings are displayed (using the data-hide-on/data-show-on attributes feature). 

2. Event dispatched to inject a request along with it's type (email_open, page_it, etc) into the enabled protocol's queue. The event object would need a checkContext method the listener can use to determine if it's protocol is enabled or not.

3. A new command that dispatches an event to consume a given queue (for example, email opens and page hits). It can use the same event object as number 2. It'll be up to the listener to communicate with it's supported server to extract and process the queue however it wants.

The idea here is to make this feature flexible and scalable without having to hard code all the support into core. A plugin providing the same features could easily be dropped into Mautic. 

I foresee a structure like this (maybe with some helper/model classes per protocol):
```
QueueBundle/
--Command/
----ConsumeQueueCommand.php
--Config/
----config.php
----rabbitmq.php
----beanstalkd.php
--DependencyInjection/
----MauticQueueExtension.php (see note below)
--Event/
----QueueEvent.php
--EventListener/
----ConfigSubscriber.php
----RabbitMqSubscriber.php
----BeanstalkdSubscriber.php
--Form/
----Type/
------ConfigType.php
------RabbitMqType.php
------BeanstalkdSubscriber.php
--Queue/
----QueueService.php (this would use the event_dispatcher to dispatch the event to inject into the listening protocol's queue instead of injecting it directly into the queue)
----QueueName.php (abstract class constants of default supported queues (email, pixel, etc)
--Views/
----FormTheme/
------Config/
--------_config_queueconfig_row.html.php
--Translations/
----en_US/
------messages.ini
------validators.ini
QueueEvents.php
MauticQueueBundle.php
```

In MauticQueueExtension.php is where the config is registered per the protocol enabled (not tested obviously):
```
<?php

/*
 * @copyright   2016 Mautic Contributors. All rights reserved
 * @author      Mautic
 *
 * @link        http://mautic.org
 *
 * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
 */

namespace Mautic\QueueBundle\DependencyInjection;

use Symfony\Component\DependencyInjection\ContainerBuilder;
use Symfony\Component\HttpKernel\DependencyInjection\Extension;

/**
 * Class MauticQueueExtension.
 *
 * This is the class that loads and manages your bundle configuration
 * To learn more see {@link http://symfony.com/doc/current/cookbook/bundles/extension.html}
 */
class MauticQueueExtension extends Extension
{
    /**
     * {@inheritdoc}
     */
    public function load(array $configs, ContainerBuilder $container)
    {
        if ($queueProtocol = $container->getParameter('mautic.queue_protocol')) {
            if (file_exists(__DIR__.'/../Config/'.$queueProtocol.'.php')) {
                include __DIR__.'/../Config/'.$queueProtocol.'.php';
            }
        }
    }
}
```

This should provide a fully flexible means to drop in additional support for other queues or pave the way for plugins via the listeners (the plugin would handle it's own config registration).